scalahost/nsc/src/main/scala/scala/meta/internal/semantic/LengthWriter.scala

```diff
@@ -0,0 +1,17 @@
+package scala.meta.internal.semantic
+
+import java.io.Writer
+
+// A writer that keeps track of the current length.
```

Comment:
No, I haven't. AFAIK, the only `Writer` that gives you a `size()`-like method is `CharArrayWriter`, but that's a final sink.